packages/electron/test/open.spec.ts

```diff
@@ -0,0 +1,439 @@
+  }
+})
+
+vi.mock('child_process', async () => {
+  return {
+    ...await vi.importActual('child_process'),
+    spawn: vi.fn(),
+  }
+})
+
+vi.mock('@packages/stderr-filtering', () => {
+  return {
+    filter: vi.fn(),
+    DEBUG_PREFIX: 'DEBUG_PREFIX',
+  }
+})
+
+vi.mock('../src/inspectArgument', () => {
+  return {
+    getInspectFromUrl: vi.fn(),
+    getInspectFromOpts: vi.fn(),
+  }
+})
+
+vi.mock('inspector')
+vi.mock('child_process')
+vi.mock('os')
+vi.mock('debug')
+describe('open', () => {
+  let argv: string[]
+  let mockChildProcess: ChildProcessWithoutNullStreams
+  const execPath = 'path/to/exec'
+  const resourcesPath = 'path/to/resources'
+  const appPath = 'path/to/app'
+  let mockProcessStdoutWritePipe: Writable
+  let mockProcessStdinWritePipe: Writable
+  let mockFilterWriter: Writable
+  let mockElectronDebugFn: Mock<ReturnType<typeof Debug>>
+  let mockStderrDebugFn: Mock<ReturnType<typeof Debug>>
+  let originalEnv: Map<string, string>
+
+  function setEnv (key: string, value: string) {
+    originalEnv.set(process.env[key] ?? '', value)
+    process.env[key] = value
+  }
+
+  function restoreEnv () {
+    for (const [key, value] of originalEnv) {
+      process.env[key] = value
+    }
+  }
+
+  beforeEach(() => {
+    originalEnv = new Map()
+    // @ts-expect-error
+    mockChildProcess = vi.mocked<ChildProcess>({
+      on: vi.fn(),
+      stderr: {
+        pipe: vi.fn(),
+      },
+      stdout: {
+        pipe: vi.fn(),
+      },
+      stdin: {
+        pipe: vi.fn(),
+      },
+    })
+
+    // @ts-expect-error
+    mockProcessStdoutWritePipe = vi.mocked<Writable>({
+      pipe: vi.fn(),
+    })
+
+    // @ts-expect-error
+    mockProcessStdinWritePipe = vi.mocked<Writable>({
+      pipe: vi.fn(),
+    })
+
+    // @ts-expect-error
+    mockFilterWriter = vi.mocked<Writable>({
+      pipe: vi.fn(),
+    })
+
+    vi.spyOn(process.stdout, 'pipe').mockReturnValue(mockProcessStdoutWritePipe)
+    vi.spyOn(process.stdin, 'pipe').mockReturnValue(mockProcessStdinWritePipe)
+
+    // happy path defaults
+    argv = ['--port', '1234']
+    vi.spyOn(inspector, 'url').mockReturnValue(undefined)
+    vi.mocked(spawn).mockReturnValue(mockChildProcess)
+    vi.mocked(access).mockResolvedValue(undefined)
+    vi.mocked(remove).mockResolvedValue(undefined)
+    vi.mocked(ensureSymlink).mockResolvedValue(undefined)
+    vi.mocked(getPathToExec).mockReturnValue(execPath)
+    vi.mocked(getPathToResources).mockReturnValue(resourcesPath)
+    vi.mocked(getSymlinkType).mockReturnValue('dir')
+    vi.mocked(filter).mockReturnValue(process.stderr)
+    mockElectronDebugFn = vi.fn()
+    mockStderrDebugFn = vi.fn()
+    // @ts-expect-error
+    vi.mocked(Debug).mockImplementation((ns) => {
+      if (ns === 'cypress:electron') {
+        return mockElectronDebugFn
+      }
+
+      if (ns === 'cypress:internal-stderr') {
+        return mockStderrDebugFn
+      }
+    })
+
+    filter.mockReturnValue(mockFilterWriter)
```

Comment:
I need to use `vi.mocked` inside the CLI. Didn't know this was a thing!

```suggestion
    vi.mocked(filter).mockReturnValue(mockFilterWriter)
```